src/DependencyManagement/DependencyManager.cs

```diff
@@ -0,0 +1,167 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.DependencyManagement
+{
+    internal class DependencyManager
+    {
+        // The list of dependent modules for the function app.
+        internal static List<DependencyInfo> Dependencies { get; private set; }
+
+        // Keep track of whether the function app dependencies have been installed.
+        internal static bool FunctionAppDependenciesInstalled { get; set; }
+
+        // Requirements.psd1 file name.
+        private const string RequirementsPsd1FileName = "Requirements.psd1";
+
+        // The list of managed dependencies supported in Azure Functions.
+        internal static readonly List<string> SupportedManagedDependencies = new List<string>(){"Az"};
+
+        internal DependencyManager()
+        {
+            Dependencies = new List<DependencyInfo>();
+        }
+
+        /// <summary>
+        /// Process Function App dependencies if specified in FunctionLoadRequest.DependencyDownloadRequest.
+        /// </summary>
+        internal void SetFunctionAppDependencies(FunctionLoadRequest request)
+        {
+            // Resolve the FunctionApp root path.
+            var functionAppRootPath = Path.GetFullPath(Path.Join(request.Metadata.Directory, ".."));
+
+            // Location where the dependent modules will be installed.
+            var modulesFolderPath = Path.Join(functionAppRootPath, "Modules");
+
+            // Path to Requirements.psd1 file.
+            var requirementsFilePath = Path.Join(functionAppRootPath, RequirementsPsd1FileName);
+
+            // Parse and process Requirements.psd1.
+            Hashtable entries = ParsePowerShellDataFile(requirementsFilePath);

```

Comment:
entries need to be checked for nullability before using it as ParsePowerShellDataFile can return null for the Hashtable

Reply:
Fixed. Thanks @SatishRanjan